src/modules/user-settings/components/tests/terminal-picker.test.tsx

```diff
@@ -1,24 +1,127 @@
-import { render, screen, fireEvent } from '@testing-library/react';
+import { Reducer, UnknownAction } from '@reduxjs/toolkit';
+import { QueryStatus } from '@reduxjs/toolkit/query';
+import { render, screen, fireEvent, waitFor } from '@testing-library/react';
+import { produce } from 'immer';
+import { Provider } from 'react-redux';
+import { getIpcApi } from 'src/lib/get-ipc-api';
 import { TerminalPicker } from 'src/modules/user-settings/components/terminal-picker';
+import { RootState, store } from 'src/stores';
+import { InstalledAppsState } from 'src/stores/installed-apps-api';
+import { testReducer } from 'src/stores/tests/utils/test-reducer';
+
+jest.mock( 'src/lib/get-ipc-api' );
+const mockGetIpcApi = getIpcApi as jest.Mock;
+
+// Create a test reducer for installedAppsApi
+function installedAppsTestReducer( state: RootState, action: UnknownAction ) {
+	if ( action.type === 'installedApps/setInstalledApps' ) {
+		const payload = action.payload as {
+			installedApps: InstalledAppsState;
+		};
+
+		return produce( state!, ( draftState ) => {
+			if ( draftState ) {
+				// Set the query result in the RTK Query cache
+				draftState.installedAppsApi.queries = {
+					'getInstalledApps({"forceRefetch":false})': {
+						status: QueryStatus.fulfilled,
+						data: payload.installedApps,
+						error: undefined,
+						requestId: 'test-request-id',
+						endpointName: 'getInstalledApps',
+						startedTimeStamp: 0,
+						fulfilledTimeStamp: 0,
+						originalArgs: undefined as never,
+					},
+				};
+			}
+		} );
+	}
+
+	return testReducer( state, action );
+}
+
+// Create test actions for installedAppsApi
+const installedAppsTestActions = {
+	setInstalledApps: ( installedApps: InstalledAppsState ) => {
+		return { type: 'installedApps/setInstalledApps', payload: { installedApps } };
+	},
+};
+
+// Replace the store's reducer with our test reducer
+store.replaceReducer( installedAppsTestReducer as Reducer< RootState > );
```

Comment:
```suggestion
store.replaceReducer( testReducer );
```
If we use the `updateQueryData` API, then it looks like `installedAppsTestReducer` is redundant.

Reply:
Thanks for the suggestion, updated in b89185c